src/rust.rs

```diff
@@ -495,6 +495,22 @@ impl HandleValue {
     }
 }
 
+impl HandleValueArray {
+    pub fn new() -> HandleValueArray {
+        HandleValueArray {
+            length_: 0,
+            elements_: ptr::null_mut(),
+        }
+    }
+
+    pub fn from_slice(values: &[Value]) -> HandleValueArray {
```

Comment:
Fair enough. Having this constructor is something I depend on quite often in the Rust Debugger API, so I'd prefer to keep it around.

I did try to make this function take a &[HandleValue] instead, since that would guarantee the values are rooted. Unfortunately, a HandleValue is not a pointer, but rather a pointer to a pointer, so you can't use it to create a HandleValueArray directly. Nor can we create a Vec<Value> from the &[HandleValue], since there's no way to keep that alive after this function returns.

Given the above, I'd like to opt to make this function unsafe.